client/src/components/utilities.ts

```diff
@@ -44,7 +44,20 @@ export const api = async <T>(
   }
 }
 
-export const apiDownload = (endpoint: string) => window.open(`/api${endpoint}`)
+export const apiDownload = (endpoint: string) =>
+  new Promise(resolve => {
+    try {
+      const windowObj = window.open(`/api${endpoint}`)
+      if (windowObj != null) {
+        windowObj.onbeforeunload = () => {
+          resolve('done')
+        }
+      }
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err)
```

Comment:
i think you should still reject the promise, otherwise wouldn't it just hang indefinitely and cause the spinner to keep showing?